lesson_13/maps_ts/src/lesson13.ts

```diff
@@ -3,5 +3,19 @@
  * https://leetcode.com/problems/permutation-difference-between-two-strings
  */
 export function findPermutationDifference(s: string, t: string): number {
-  return 0;
+  const map = new Map<string, number>();
+
+  for (let i = 0; i < t.length; i++) {
+    map.set(t[i], i);
+  }
+
+  let difference = 0;
+  for (let i = 0; i < s.length; i++) {
+    const idx = map.get(s[i]);
+    if (idx != undefined) {
```

Comment:
Another way to do this is using the null coalesce operator `??`:

```javascript
difference += Math.abs(i - idx ?? i);
```

Reply:
Interesting why is typescript so weird